### PR TITLE
HPCC-30413 Add option to always capture post-mortem info

### PR DESCRIPTION
**Pull request by jakesmith**
Enabled either globally (i.e. for all components), or for individual components (e.g. Thor).
Usage:
```
expert:
  alwaysPostMortem: true
```

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -689,13 +689,13 @@ Check that the storage and spill planes for a component exist
 Add command for a component
 */}}
 {{- define "hpcc.componentCommand" -}}
-{{- if .me.valgrind -}}
+ {{- if .me.valgrind -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:692]: We don't normally indent the root statements inside a define.  Was this deliberate?

File: helm/hpcc/templates/_helpers.tpl [hunk 2]
```diff
@@ -713,6 +713,9 @@ Add extra args for a component
  {{- $debugPlane := .me.debugPlane | default (include "hpcc.getFirstPlaneForCategory"  (dict "root" .root "category" "debug")) -}}
  {{- include "hpcc.checkPlaneExists" (dict "root" .root "planeName" $debugPlane) -}}
  {{- $prefix := include "hpcc.getPlanePrefix" (dict "root" .root "planeName" $debugPlane) -}}
+  {{- if or (and (hasKey .me "expert") .me.expert.alwaysPostMortem) (and (hasKey .root.Values.global "expert") .root.Values.global.expert.alwaysPostMortem) -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:716]: This isn't short circuited on old versions of helm. 
MeExpert := .me.expert | dict
?

**Review by ghalliday — COMMENTED** [review]
Functionality looks good.  A couple of helm comments.

File: helm/hpcc/templates/_helpers.tpl (hunk 2 above)
Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl:716]: hm, yes good point, will change.

File: helm/hpcc/templates/_helpers.tpl (hunk 1 above)
Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl:692]: you're right we don't (it looks clearer, but..), I'll revert.

**Comment by jakesmith** [comment]
@ghalliday - see 2nd commit.

**Review by ghalliday — CHANGES_REQUESTED** [review]
@jakesmith the csv change should be a different PR - especially switching output over to using it.

File: system/jlib/jlog.cpp [hunk 3]
```diff
@@ -879,7 +956,7 @@ void CategoryLogMsgFilter::reset()
 void HandleLogMsgHandlerTable::handleMessage(const LogMsg & msg)
 {
     CriticalBlock block(crit);
-    msg.toStringTable(curMsgText.clear(), messageFields);
+    msg.toStringCSV(curMsgText.clear(), messageFields);
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply system/jlib/jlog.cpp:959]: whoops, I did not.

**Comment by jakesmith** [comment]
> @jakesmith the csv change should be a different PR - especially switching output over to using it.

rollback unintentional jlog changes - see 3rd commit.

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash

**Comment by jakesmith** [comment]
@ghalliday - squashed

**Pull request merged**